src/ORM/DataQuery.php

```diff
@@ -912,6 +914,14 @@ protected function joinHasManyRelation(
             return;
         }
 
+        /** @var Versioned $versioned */
+        $versioned = Injector::inst()->get(Versioned::class);
```

Comment:
`Versioned` is technically a dev requirement so we can't assume that it's installed. Preferably this would actually live in Versioned code too. Perhaps we can just put

```
$this->extend('updateForeignTable', $foreignTable, $foreignClass);
```

And then have this code in `Versioned`?

Reply:
I think that's fine, will try to pick it up on the next hack day, but I don't know if I have time. Do you think this requires someone else to pick it up and fix the issue earlier?

Reply:
I will need to get more familiar with the bug to give you a good answer. I'm happy to help out when you next pick it up though :)